### PR TITLE
[CI:DOCS] docs: update network tutorial with netavark DHCP support

### LINKED ISSUE #17635: Basic networking tutorial incorrect podman create network command on MACVLAN needs updating
### Issue Description

Issue Description: https://github.com/containers/podman/blob/main/docs/tutorials/basic_networking.md#macvlan MACVLAN tutorial section has an incorrect command when creating the macvlan network.

### Steps to reproduce the issue

Steps to reproduce the issue:
1.Follow the tutorial for basic networking (macvlan section) https://github.com/containers/podman/blob/main/docs/tutorials/basic_networking.md#macvlan
2.  Issue the command `sudo podman network create -d macvlan -o parent=eno1 webnetwork`

### Describe the results you received

Describe the results you received:
Error: macvlan driver needs at least one subnet specified, DHCP is not yet supported with netavark

### Describe the results you expected

Describe the results you expected

### podman info output

```yaml
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.

> podman info
host:
  arch: amd64
  buildahVersion: 1.29.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon is owned by conmon 1:2.1.6-1
    path: /usr/bin/conmon
    version: 'conmon version 2.1.6, commit: 158b5421dbac6bda96b1457955cf2e3c34af29bc'
  cpuUtilization:
    idlePercent: 98.43
    systemPercent: 1.1
    userPercent: 0.47
  cpus: 4
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: arch
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 6.1.12-arch1-1
  linkmode: dynamic
  logDriver: journald
  memFree: 15233699840
  memTotal: 16664723456
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 1.8-1
    path: /usr/bin/crun
    version: |-
      crun version 1.8
      commit: 0356bf4aff9a133d655dc13b1d9ac9424706cac4
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns is owned by slirp4netns 1.2.0-1
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.4
  swapFree: 2147479552
  swapTotal: 2147479552
  uptime: 1h 53m 59.00s (Approximately 0.04 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/iand/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/iand/.local/share/containers/storage
  graphRootAllocated: 499568873472
  graphRootUsed: 5012037632
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/iand/.local/share/containers/storage/volumes
version:
  APIVersion: 4.4.2
  Built: 1677255177
  BuiltTime: Sat Feb 25 03:12:57 2023
  GitCommit: 74afe26887f814d1c39925a1624851ef3590e79c-dirty
  GoVersion: go1.20.1
  Os: linux
  OsArch: linux/amd64
  Version: 4.4.2
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

No

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

happened after reseting podman to learn the network stack

### PR DESCRIPTION
**Pull request by Luap99**
Add instructions on how to start the netavark dhcp proxy. Also list version requirements.

Fixes #17635

#### Does this PR introduce a user-facing change?

```release-note
Updated the basic_networking tutorial with netavark DHCP support.
```

### CODE REVIEW

**Comment by Luap99** [comment]
@baude @mheon PTAL

File: docs/tutorials/basic_networking.md [hunk 1]
```diff
@@ -167,23 +167,53 @@ is leased from a DHCP server like most other network clients on the network.  If
...
 ```
 $ sudo podman network create -d macvlan -o parent=eth0 webnetwork
 webnetwork
 ```
-The next step is to ensure that the DHCP CNI plugin is running.  This plugin facilitates
-the DHCP lease from the network.
+
+The next step is to ensure that the DHCP service is running. This handles
+the DHCP leases from the network. If DHCP is not needed the `--subnet` option
+can be used to assign a static subnet in network create command above.
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/basic_networking.md:186]:
```suggestion
can be used to assign a static subnet in the `network create` command above.
```

File: docs/tutorials/basic_networking.md [hunk 2]
```diff
@@ -167,23 +167,53 @@ is leased from a DHCP server like most other network clients on the network.  If
...
 
 ```
 $ sudo podman network create -d macvlan -o parent=eth0 webnetwork
 webnetwork
 ```
-The next step is to ensure that the DHCP CNI plugin is running.  This plugin facilitates
-the DHCP lease from the network.
+
+The next step is to ensure that the DHCP service is running. This handles
+the DHCP leases from the network. If DHCP is not needed the `--subnet` option
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/basic_networking.md:185]:
```suggestion
the DHCP leases from the network. If DHCP is not needed, the `--subnet` option
```

File: docs/tutorials/basic_networking.md [hunk 3]
```diff
@@ -167,23 +167,53 @@ is leased from a DHCP server like most other network clients on the network.  If
...
+can be used to assign a static subnet in network create command above.
+
+CNI and netavark use both their own dhcp service, therefore you need to know
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/basic_networking.md:188]:
```suggestion
CNI and netavark both use their own DHCP service; therefore you need to know
```

File: docs/tutorials/basic_networking.md [hunk 4]
```diff
@@ -167,23 +167,53 @@ is leased from a DHCP server like most other network clients on the network.  If
...
+what backend you are using. To see what you are using run this command:
+```
+$ sudo podman info --format {{.Host.NetworkBackend}}
+```
+If this command does not work you are using an older version before v4.0
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/basic_networking.md:193]:
```suggestion
If this command does not work, you are using an older version prior to Podman v4.0
```

File: docs/tutorials/basic_networking.md [hunk 5]
```diff
@@ -167,23 +167,53 @@ is leased from a DHCP server like most other network clients on the network.  If
...
+which means you are using CNI.
+If the netavark backend is used at least Podman v4.5 with netavark v1.6 is
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/basic_networking.md:195]:
```suggestion
If the netavark backend is used, at least Podman v4.5 with netavark v1.6 is
```

File: docs/tutorials/basic_networking.md [hunk 6]
```diff
@@ -167,23 +167,53 @@ is leased from a DHCP server like most other network clients on the network.  If
...
+
+With CNI use:
+```
+$ sudo systemctl enable --now cni-dhcp.socket
+```
+Or if the system doesn't use systemd, start the daemon manually:
 ```
 $ sudo /usr/libexec/cni/dhcp daemon
 ```
+Note that depending on the distribution the binary location may differ.
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/basic_networking.md:215]:
```suggestion
Note that depending on the distribution, the binary location may differ.
```

**Comment by TomSweeneyRedHat** [comment]
Overall it looks good, just a few sprinkled nits.

**Comment by rhatdan** [comment]
Has this been covered in a Blog?  I think a new blog on setting up netavark+dns+dhcp would be useful.  Tutorials are great, but not in a prominent place.  Taking this tutorial and translating it into a new blog would not be that much work.

**Comment by Luap99** [comment]
> Has this been covered in a Blog? I think a new blog on setting up netavark+dns+dhcp would be useful. Tutorials are great, but not in a prominent place. Taking this tutorial and translating it into a new blog would not be that much work.

Not yet but I think this would be useful. This is just to make sure the docs in this repo are at least accurate.

**Comment by Luap99** [comment]
Thanks @TomSweeneyRedHat, I applied all your suggestions.

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons.  A blog on this would be some serious awesome-sauce.

**Pull request merged**